File: C005 Driving Code Examples/004 LinkedList.py
###########Link source###############


#############Python Version###########

# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next

 # Node Class
class Node:
   
    # Constructor to create a new node
    def __init__(self, d):
        self.data = d
        self.next = None

    
head = None
head = Node(1)
head.next = Node(2)
head.next.next = Node(3)
head.next.next.next = Node(4)
head.next.next.next.next = Node(5)

File: C005 Driving Code Examples/004 LinkedList.java
public static void main(String[] args)
    {
 
        LinkedList list = new LinkedList();
        list.head = new Node(1);
        list.head.next = new Node(2);
        list.head.next.next = new Node(3);
        list.head.next.next.next = new Node(4);
        list.head.next.next.next.next = new Node(5);
 
        list.printlist(head); // print original list
        list.rearrange(head); // rearrange list as per ques
        System.out.println("");
        list.printlist(head); // print modified list
    }
}